### PR TITLE
add sparse_map

### PR DESCRIPTION
**Pull request by GlynLeine**
[Add] sparse_map.hpp

### CODE REVIEW

File: applications/test_game/source.cpp [hunk 1]
```diff
@@ -1,11 +1,85 @@
...
+
 void ARGS_CCONV reportModules(args::core::Engine* engine)
 {
 	std::cout << "Hello Args!" << std::endl;
 	engine->reportModule<TestModule>();
+
+	std::cout << "_____________________________________________________" << std::endl;
+
+
+	args::core::sparse_map<int, sah> testMap;
```
Comment (Algo-ryth-mix) [inline applications/test_game/source.cpp:21]: shouldn't a sparse set only hold uint ? 🤔

**Review by Algo-ryth-mix — APPROVED** [review]

File: applications/test_game/source.cpp (hunk 1 above)
Comment (GlynLeine) [reply applications/test_game/source.cpp:21]: normally yes, but by using a hash map as the sparse container and storing a pair with the key and the value in the dense container we can map any type value to any other type value and using the concept of a sparse set inbetween

Comment (GlynLeine) [reply applications/test_game/source.cpp:21]: the traditional sparse_set is still planned and will come either tomorrow or monday prolly

Comment (Algo-ryth-mix) [reply applications/test_game/source.cpp:21]: fine by me. there is an implementation of sparse_set in lpe if you want a pretty much ready made one

**Pull request merged**